tests/parser/test-block-bindings.php

```diff
@@ -0,0 +1,280 @@
+<?php
+/**
+ * Class BlockBindingsTest
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+use WP_Block;
+
+/**
+ * Test parsing blocks with block binding.
+ */
+class BlockBindingsTest extends RegistryTestCase {
+	protected function setUp(): void {
+		parent::setUp();
+
+		if ( ! class_exists( 'WP_Block_Bindings_Registry' ) ) {
+			$this->markTestSkipped( 'This test suite requires WP_Block_Bindings_Registry (WordPress 6.5 or higher).' );
+		}
+	}
+
+	/* Single paragraph block binding */
+
+	public function test_single_paragraph_block_binding() {
+
+		$this->register_block_bindings_source(
+			'test/block-binding',
+			[
+				'label'              => 'Test paragraph block binding',
+				'get_value_callback' => static function ( array $args, WP_Block $block ) {
+					return sprintf( 'Block binding for %s with arg foo=%s', $block->name, $args['foo'] );
+				},
+			]
+		);
+
+		$html = '
+			<!-- wp:core/paragraph {"metadata":{"bindings":{"content":{"source":"test/block-binding","args":{"foo":"bar"}}}}} -->
+			<p>Fallback content</p>
+			<!-- /wp:core/paragraph -->
+		';
+
+		$expected_blocks = [
+			[
+				'name'       => 'core/paragraph',
+				'attributes' => [
+					'content'  => 'Block binding for core/paragraph with arg foo=bar',
+					'metadata' => [
+						'bindings' => [
+							'content' => [
+								'source' => 'test/block-binding',
+								'args'   => [ 'foo' => 'bar' ],
+							],
+						],
+					],
+				],
+			],
+		];
+
+		$content_parser = new ContentParser( $this->get_block_registry() );
+		$blocks         = $content_parser->parse( $html );
+
+		$this->assertArrayHasKey( 'blocks', $blocks, sprintf( 'Unexpected parser output: %s', wp_json_encode( $blocks ) ) );
+
+		// Because this test uses a core block, we use assertArraySubset to avoid brittle
```

Comment:
Could you expand this comment to include why you are using a core block? My understanding is that `core/paragraph` is special and custom blocks do not support content bindings, but it could also just be because you wanted to test a core block.

Reply:
It's the former, block bindings are not yet supported outside of a few core blocks. Clarified in https://github.com/Automattic/vip-block-data-api/pull/74/commits/a67d7e5efd8aa402c97b7d56cf0e15fc1f32433f